server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -242,10 +264,10 @@ static ClusterState addIndexClosedBlocks(final Index[] indices, final Map<Index,
                 blocks.removeIndexBlockWithId(index.getName(), INDEX_CLOSED_BLOCK_ID);
                 routingTable.remove(index.getName());
                 indexBlock = INDEX_CLOSED_BLOCK;
-            }
-            if (indexBlock == null) {
+            } else if (indexBlock == null) {
                 // Create a new index closed block
-                indexBlock = createIndexClosedBlock();
+                indexBlock = createIndexClosingBlock();
+                assert Strings.hasLength(indexBlock.uuid()) : "Closing block should have a UUID";
```

Comment:
this assertion should also cover the case where `indexBlock` was not previously null, i.e., rewrite this as:

```
} else {
  if (indexBlock == null) {
    indexBlock = createIndexClosingBlock();
  }
  assert Strings.hasLength(indexBlock.uuid()) : "Closing block should have a UUID";
}
```

Reply:
Sure